packages/solarwinds-apm/test/appoptics/sampler.test.ts

```diff
@@ -0,0 +1,249 @@
+      {
+        "X-Trace-Options": "trigger-trace;ts=12345678",
+        "X-Trace-Options-Signature": "signature",
+      },
+      { triggerTrace: true, timestamp: 12345678, custom: {}, ignored: [] },
+    )
+
+    expect(options).to.loosely.deep.equal({
+      in_xtrace: null,
+      custom_trigger_mode: oboe.TRIGGER_ENABLED,
+      request_type: oboe.REQUEST_TYPE_TRIGGER,
+      header_options: "trigger-trace;ts=12345678",
+      header_signature: "signature",
+      header_timestamp: 12345678,
+    } satisfies oboe.DecisionOptions)
+  })
+
+  lit("creates proper options for custom transaction settings spans", () => {
+    const options = intoOboeDecisionOptions(
+      undefined,
+      SpanType.ROOT,
+      { triggerMode: false, tracingMode: TracingMode.NEVER },
+      {},
+      undefined,
+    )
+
+    expect(options).to.loosely.deep.equal({
+      in_xtrace: null,
+      custom_tracing_mode: oboe.TRACE_DISABLED,
+      custom_trigger_mode: oboe.TRIGGER_DISABLED,
+      request_type: oboe.REQUEST_TYPE_REGULAR,
+    } satisfies oboe.DecisionOptions)
+  })
+})
+
+describe("fromOboeDecisionsResult", () => {
+  lit("creates proper result for sampled span", () => {
+    const result = fromOboeDecisionsResult(
+      {
+        do_sample: 1,
+        do_metrics: 1,
+        sample_rate: 50_000,
+        bucket_cap: 10,
+        bucket_rate: 100,
+        type: oboe.SETTINGS_TYPE_LAYER_SAMPLE_RATE,
+        sample_source: oboe.SAMPLE_RATE_SOURCE_OBOE,
+        status: oboe.TRACING_DECISIONS_OK,
+        status_msg: "ok",
+        auth: oboe.TRACING_DECISIONS_AUTH_NOT_PRESENT,
+        auth_msg: "",
+      },
+      undefined,
+    )
+
+    expect(result).to.loosely.deep.equal({
+      samplingResult: {
+        decision: SamplingDecision.RECORD_AND_SAMPLED,
+        attributes: {
+          SampleRate: 50_000,
+          BucketCapacity: 10,
+          BucketRate: 100,
+        },
+      },
+      traceOptionsResponse: {},
+    })
+  })
+
+  lit("creates proper result for trigger-traced span", () => {
+    const result = fromOboeDecisionsResult(
+      {
+        do_sample: 1,
+        do_metrics: 1,
+        sample_rate: 100_000,
+        bucket_cap: 100,
+        bucket_rate: 500,
+        type: oboe.SETTINGS_TYPE_LAYER_SAMPLE_RATE,
+        sample_source: oboe.SAMPLE_RATE_SOURCE_OBOE,
+        status: oboe.TRACING_DECISIONS_OK,
+        status_msg: "ok",
+        auth: oboe.TRACING_DECISIONS_AUTH_OK,
+        auth_msg: "ok",
+      },
+      {
+        triggerTrace: true,
+        timestamp: 12345678,
+        custom: { "custom-foo": "custom-bar" },
+        ignored: [["foo", "bar"]],
+        swKeys: "sw",
+      },
+    )
+
+    expect(result).to.loosely.deep.equal({
+      samplingResult: {
+        decision: SamplingDecision.RECORD_AND_SAMPLED,
+        attributes: {
+          SampleRate: 100_000,
+          BucketCapacity: 100,
+          BucketRate: 500,
+          "custom-foo": "custom-bar",
+          SWKeys: "sw",
```

Comment:
Would the `TriggeredTrace: true` [attribute](https://swicloud.atlassian.net/wiki/spaces/NIT/pages/2770436395/Trigger+Trace#X-Trace-Options) also be set here and if so, can the test also check? 

Also, pretty minor, the [SampleRate and SampleSource](https://github.com/solarwinds-cloud/trace/blob/master/docs/specs/KV/service-entry-internal.md) ideally would be unset in the case of a triggered trace.

Reply:
I was not aware of the `TriggeredTrace` attribute, I think it was never implemented here woops. I'll add those changes, I also need to change some of the stable HTTP semconv handling because `url.full` isn't actually present on the spans

Reply:
Updated to add the missing `TriggeredTrace` attribute and also `SampleSource` which was also missing. Also removed any logic related to local sample rate since that made implementing sample sources easier and it's not used anywhere anyway.